src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+export enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+function filteringTodos(
+  array: Todo[],
+  settings: { filterType: Filter; query: string },
+) {
+  let arrayCopy = [...array];
+
+  if (settings.filterType === Filter.All) {
+    arrayCopy = [...array];
+  }
+
+  if (settings.filterType === Filter.Active) {
+    arrayCopy = arrayCopy.filter(element => element.completed === false);
+  }
+
+  if (settings.filterType === Filter.Completed) {
+    arrayCopy = arrayCopy.filter(element => element.completed === true);
+  }
+
+  if (settings.query) {
+    arrayCopy = arrayCopy.filter(element =>
+      element.title
+        .toLowerCase()
+        .includes(settings.query.toLowerCase().trimStart()),
+    );
+  }
+
+  return arrayCopy;
+}
```

Comment:
The function `filteringTodos` is not pure. It creates a new array copy on every call, which is not efficient. Also, it's not necessary to create a copy of the array if the filter type is `All`. Instead, you can return the original array. This will improve the performance of your application.